cmd/autobrrctl/main.go

```diff
@@ -342,3 +416,107 @@ func CreateHtpasswdHash(password string) (string, error) {
+		"created_at", "updated_at", // Timestamps not in WebUI export data
+		"priority", "smart_episode", // Other fields not in WebUI export data (use JSON names)
+		"actions_count", "actions_enabled_count", // Derived/extra fields if present
+	}
+	for _, key := range fieldsToRemove {
+		delete(filterMap, key)
+	}
+
+	// Fields to remove if they have default values (mirroring frontend logic)
+	// Note: JSON unmarshals numbers as float64 and empty arrays as []interface{}
+	defaults := map[string]interface{}{
+		"enabled":        false,
+		"matchReleases":  []interface{}{},
+		"exceptReleases": []interface{}{},
+		"tags":           []interface{}{},
+		"categories":     []interface{}{},
+		"resolutions":    []interface{}{},
+		"source":         []interface{}{},
+		"type":           []interface{}{},
+		"codecs":         []interface{}{},
+		"container":      []interface{}{},
+		"freeleech":      []interface{}{},
+		"searchType":     []interface{}{},
+		"searchEngine":   []interface{}{},
+		"matchTorrents":  true,
+		"episodeFilter":  float64(0),
+		"seasonFilter":   float64(0),
+		"smartFilter":    false,
+		// Add other fields with defaults if needed
+	}
+
+	for key, defaultValue := range defaults {
+		if value, ok := filterMap[key]; ok {
+			// Use reflect.DeepEqual for robust comparison, especially for slices
+			if reflect.DeepEqual(value, defaultValue) {
+				delete(filterMap, key)
+			} else if reflect.TypeOf(value).Kind() == reflect.Slice && reflect.ValueOf(value).Len() == 0 {
+				// Handle case where default is empty slice literal and value is non-nil empty slice
+				if reflect.DeepEqual(defaultValue, []interface{}{}) {
+					delete(filterMap, key)
+				}
+			}
+		}
+	}
+
+	// Add external filters if any exist
+	if len(externalFilters) > 0 {
+		// Marshal external filters to add them correctly structured
+		externalData, err := json.Marshal(externalFilters)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal external filters: %w", err)
+		}
+		var externalMapInterface interface{} // Use interface{} to handle potential array/object variations
+		if err := json.Unmarshal(externalData, &externalMapInterface); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal external filters to map: %w", err)
+		}
+		filterMap["external"] = externalMapInterface
+	} else {
+		// Ensure 'external' key doesn't exist if there are no external filters
+		// This prevents `"external": null` if the field exists on the original struct
+		delete(filterMap, "external")
+	}
+
+	// Remove the name field from the data map as it's already at the root level
+	delete(filterMap, "name")
+
+	// Create the final output structure
+	outputMap := map[string]interface{}{
+		"name":    filter.Name, // Use the original filter name
+		"version": "1.0",       // Match WebUI version format
+		"data":    filterMap,   // Nest the cleaned filter data here
+	}
+
+	// Marshal the cleaned map back to JSON with indentation
+	return json.MarshalIndent(outputMap, "", "  ")
+}
```

Comment:
Any reason this isn't a new struct type with just the fields we want? Other than having to keep them in sync when we update the Filters, I don't see much reason to use a map here 🤔

Reply:
TBH I don't know why is the WebUI removing certain fields and changing the structure? Let me try to improve this as per your comment.